### PR TITLE
Remove the Rackspace provider

### LINKED ISSUE #6962: Consider removing Rackspace support
Rackspace support has not ben updated in a few months.
cluster/rackspace/cloud-config/minion-cloud-config.yaml is using etcd_servers for kubelet and kube-proxy, but support for them has been removed since v0.9, I think.

I'll remove it in a few days if no-one makes a counter-offer to get the config working with latest release.

### PR DESCRIPTION
**Pull request by etoews**
**What this PR does / why we need it**:

To aid the effort of moving providers out of the cluster dir, I'm
removing Rackspace and leaving behind a README.md simply as a
placeholder until the entire dir is deleted.

**Which issue this PR fixes**

Fixes #6962

**Release note**:

```release-note
Deployment of Kubernetes clusters on Rackspace using the in-tree bash deployment (i.e. cluster/kube-up.sh or get-kube.sh) is obsolete and support has been removed.```

### CODE REVIEW

**Comment by kenperkins** [comment]
Looks good to me.

**Comment by roberthbailey** [comment]
@everett-toews -- I updated the release note to indicate the removal of in-tree bash support for rackspace. If there are alternative setup guides that we can point people to, please add those to the release note as well.

File: cluster/rackspace/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Rackspace
```
Comment (roberthbailey) [inline cluster/rackspace/README.md]: I don't think that this file is necessary. We are trying to clear out the whole cluster directory, so the absence of a rackspace subdirectory won't seem strange.

**Comment by everett-toews** [comment]
@roberthbailey This is RFR again. Currently there's no alternative setup guide so no need to modify the release note any further.

**Comment by roberthbailey** [comment]
please squash your commits then lgtm. 

@mikedanese can approve in my absence once you squash.

**Comment by everett-toews** [comment]
Squashed. Thanks!

**Comment by everett-toews** [comment]
@k8s-bot unit test this

**Comment by everett-toews** [comment]
Hrm. Those test fails are unrelated to this PR and seem like they're an intermittent error. I don't have the perms to retest this so I'm not entirely sure how to proceed.

@mikedanese Any advice?

**Comment by mikedanese** [comment]
Caused by https://github.com/kubernetes/kubernetes/issues/38228#issuecomment-298039921

**Comment by mikedanese** [comment]
@k8s-bot unit test this
@k8s-bot gce etcd3 e2e test this

**Comment by everett-toews** [comment]
@mikedanese Thank you!

**Pull request merged**